### PR TITLE
issue-524 change recipients color immediatly if they were added to co…

### PR DESCRIPTION
**Pull request by michael-volynets**
…ntacts

#524

Issue #524 
Time: ~4 hrs
It took 2h  for investigating how BrowserMsg works.

### CODE REVIEW

**Comment by tomholub** [comment]
Please also check email I've sent you a few days ago, to remember to note how much time each PR took. Thanks!

**Comment by michael-volynets** [comment]
@tomholub  Yes, I've done that

File: extension/js/common/composer.ts [hunk 1]
```diff
@@ -436,6 +438,17 @@ export class Composer {
     }
   }
 
+  private checkReciepientsKeys = async () => {
+    for (const recipientEl of $('.recipients span.no_pgp')) {
+      const email = $(recipientEl).text().trim();
+      const res = await this.lookupPubkeyFromDbOrKeyserverAndUpdateDbIfneeded(email);
```
Comment (tomholub) [inline extension/js/common/composer.ts]: Here I recommend to check db directly, without checking attester. After all, the imported key should by now already be in the local db.

(as implemented now, If you have 3 recipients who are all gray, and add one pubkey, two of them will be re-checked on Attester unnecessarily)

Something like (please test):

```js
const [dbContact] = await this.app.storageContactGet([email]);
if(dbContact) {
  $(recipientEl).removeClass('no_pgp');
  await this.renderPubkeyResult(recipientEl, email, dbContact);
}
```

**Comment by tomholub** [comment]
> It took 2h for investigating how BrowserMsg works.

It sometimes takes me longer than that, especially when I need to make changes to it :smile:  Thanks for the note.

**Pull request merged**